src/vulkan-renderer/wrappers/instance.cpp

```diff
@@ -0,0 +1,174 @@
+#include "inexor/vulkan-renderer/wrappers/instance.hpp"
+#include <GLFW/glfw3.h>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+Instance::Instance(Instance &&other) noexcept : instance(std::exchange(other.instance, nullptr)) {}
+
+Instance::Instance(const std::string &application_name, const std::string &engine_name, const std::uint32_t application_version,
+                   const std::uint32_t engine_version, const std::uint32_t vulkan_api_version,
+                   std::optional<std::vector<std::string>> requested_instance_extensions, std::optional<std::vector<std::string>> requested_instance_layers,
+                   bool enable_validation_layers, bool enable_renderdoc_instance_layer) {
+    assert(!application_name.empty());
+    assert(!engine_name.empty());
+
+    // In Vulkan API we can use VK_MAKE_VERSION() macro to create an std::uint32_t value from major, minor and patch number.
+
+    // Get the major, minor and patch version of the application.
+    std::uint32_t app_major = VK_VERSION_MAJOR(application_version);
+    std::uint32_t app_minor = VK_VERSION_MINOR(application_version);
+    std::uint32_t app_patch = VK_VERSION_PATCH(application_version);
+
+    // Get the major, minor and patch version of the engine.
+    std::uint32_t engine_major = VK_VERSION_MAJOR(engine_version);
+    std::uint32_t engine_minor = VK_VERSION_MINOR(engine_version);
+    std::uint32_t engine_patch = VK_VERSION_PATCH(engine_version);
+
+    // Get the major, minor and patch version of the requested Vulkan API interface.
+    std::uint32_t vulkan_api_major = VK_VERSION_PATCH(vulkan_api_version);
+    std::uint32_t vulkan_api_minor = VK_VERSION_PATCH(vulkan_api_version);
+    std::uint32_t vulkan_api_patch = VK_VERSION_PATCH(vulkan_api_version);
+
+    spdlog::debug("Initialising Vulkan instance.");
+    spdlog::debug("Application name: '{}'", application_name.c_str());
+    spdlog::debug("Application version: {}.{}.{}", app_major, app_minor, app_patch);
+    spdlog::debug("Engine name: '{}'", engine_name.c_str());
+    spdlog::debug("Engine version: {}.{}.{}", engine_major, engine_minor, engine_patch);
+    spdlog::debug("Requested Vulkan API version: {}.{}.{}", vulkan_api_patch, vulkan_api_minor, vulkan_api_patch);
```

Comment:
Two times `vulkan_api_patch`.

Reply:
I just knew I would make this mistake :D.

Reply:
done.

Reply:
Also, this was wrong as well.


```
// Get the major, minor and patch version of the requested Vulkan API interface.
std::uint32_t vulkan_api_major = VK_VERSION_PATCH(vulkan_api_version);
std::uint32_t vulkan_api_minor = VK_VERSION_PATCH(vulkan_api_version);
std::uint32_t vulkan_api_patch = VK_VERSION_PATCH(vulkan_api_version);
```

It should be:

```
// Get the major, minor and patch version of the requested Vulkan API interface.
std::uint32_t vulkan_api_major = VK_VERSION_MAJOR(vulkan_api_version);
std::uint32_t vulkan_api_minor = VK_VERSION_MINOR(vulkan_api_version);
std::uint32_t vulkan_api_patch = VK_VERSION_PATCH(vulkan_api_version);
```